### PR TITLE
Prevent MultiSig actor lockout and a takeover

### PR DESCRIPTION
**Pull request by Kubuxu**
When removing Singers and changing the Required number of approvals it was previously possible to lockout the MultiSig.

The introduction of `increaseReq` is to prevent a takeover when the number of Signers is increased from 1.
Without it, there is no way to safely add another signer to a MultiSig with just one Signer.

Also, retain return code of called methods.

Removed `created` as it wasn't being set.

### CODE REVIEW

File: actors.md [hunk 1]
```diff
@@ -1440,7 +1440,7 @@ func Propose(to Address, value TokenAmount, method String, params Bytes) UInt {
 	self.Transactions.Append(tx)
 
 	if self.Required == 1 {
-		vm.Send(tx.To, tx.Value, tx.Method, tx.Params)
+		tx.RetCode := vm.Send(tx.To, tx.Value, tx.Method, tx.Params)
```
Comment (whyrusleeping) [inline actors.md]:
```suggestion
		tx.RetCode = vm.Send(tx.To, tx.Value, tx.Method, tx.Params)
```

File: actors.md [hunk 2]
```diff
@@ -1559,19 +1559,23 @@ func ClearCompleted() {
...
     signer Address
+    increaseReq bool
 } representation tuple
 ```
 
 **Algorithm**
 
 ```go
-func AddSigner(signer Address) {
+func AddSigner(signer Address, incraseReq bool) {
```
Comment (whyrusleeping) [inline actors.md]:
```suggestion
func AddSigner(signer Address, increaseReq bool) {
```

File: actors.md [hunk 3]
```diff
@@ -1597,6 +1601,10 @@ func RemoveSigner(signer Address) {
 	if !self.isSigner(signer) {
 		Fatal("given address was not a signer")
 	}
+	if len(self.Signers)-1 < self.Required {
```
Comment (dignifiedquire) [inline actors.md]: why not have an option for reducing required as well?

Comment (Kubuxu) [reply actors.md]: Added.

File: actors.md [hunk 4]
```diff
@@ -1652,6 +1661,9 @@ func ChangeRequirement(req UInt) {
 	if req < 1 {
 		Fatal("requirement must be at least 1")
 	}
+	if req > len(self.Signers) {
+		Fatal("requirement must be less than nuber of signers")
```
Comment (dignifiedquire) [inline actors.md]: typo nuber

Comment (Kubuxu) [reply actors.md]:
```suggestion
		Fatal("requirement must be less than number of signers")
```

**Pull request merged**